src/algebra/gcd_monoid/integrally_closed.lean

```diff
@@ -0,0 +1,57 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import algebra.gcd_monoid.basic
+import ring_theory.integrally_closed
+import ring_theory.polynomial.scale_roots
+import ring_theory.polynomial.eisenstein
+
+/-!
+
+# GCD domains are integrally closed
+
+-/
+
+open_locale big_operators polynomial
+
+lemma is_localization.surj_of_gcd_domain {R : Type*} [comm_ring R] [is_domain R] [gcd_monoid R]
+  (M : submonoid R) {K} [comm_semiring K] [algebra R K] [is_localization M K] (z : K) :
+  ∃ a b : R, is_unit (gcd a b) ∧ z * algebra_map R K b = algebra_map R K a :=
```

Comment:
I thought this could go into an earlier file but I couldn't find one that knows both `gcd_monoid` and `is_localization`, so it should probably just stay here.

Reply:
I figured so too. Besides, this file falls in the folder of `gcd_monoid` so I don't think it is that bad of a choice.